src/map/atcommand.cpp

```diff
@@ -2273,8 +2273,17 @@ ACMD_FUNC(monster)
 		return -1;
 	}
 
-	if ((mob_id = mobdb_searchname(monster)) == 0) // check name first (to avoid possible name begining by a number)
-		mob_id = mobdb_checkid(atoi(monster));
+	// If AegisName matches exactly, summon that monster
+	std::shared_ptr<s_mob_db> mob = mobdb_search_aegisname(monster);
+	if (mob != nullptr)
+		mob_id = mob->id;
+	else {
+		// Otherwise, search for monster with that ID or name
+		// Check for ID first as this is faster; if search string is not a number it will return 0
+		mob_id = mobdb_checkid(util::strtoint32def(monster));
+		if (mob_id == 0)
```

Comment:
```suggestion
		mob_id = util::strtoint32def(monster);
		if (mob_id == 0 || mobdb_checkid(mob_id) == 0)
```
This prevents useless mob_db.exists and mob_is_clone calls.